src/Microsoft.Tye.Core/DockerCompose/DockerComposeParser.cs

```diff
@@ -0,0 +1,455 @@
+                        break;
+                    case "privileged":
+                        break;
+                    case "read_only":
+                        break;
+                    case "shm_size":
+                        break;
+                    case "stdin_open":
+                        break;
+                    case "tty":
+                        break;
+                    default:
+                        throw new TyeYamlException(child.Key.Start, CoreStrings.FormatUnrecognizedKey(key));
+                }
+            }
+        }
+
+        private static void ParseEnvironment(YamlNode node, ConfigService service)
+        {
+            if (node is YamlSequenceNode sequenceNode)
+            {
+                foreach (var arg in sequenceNode)
+                {
+                    var configItem = new ConfigConfigurationSource();
+                    var argString = YamlParser.GetScalarValue(arg);
+                    if (argString.Contains('='))
+                    {
+                        var parts = argString.Split('=');
+                        configItem.Name = parts[0];
+                        configItem.Value = parts[1];
+                    }
+                    else
+                    {
+                        configItem.Name = argString;
+                    }
+                    service.Configuration.Add(configItem);
+                }
+            }
+            else
+            {
+                var mappingNode = (node as YamlMappingNode)!;
+                foreach (var arg in mappingNode)
+                {
+                    var configItem = new ConfigConfigurationSource();
+                    configItem.Name = YamlParser.GetScalarValue(arg.Key);
+                    configItem.Value = YamlParser.GetScalarValue(arg.Value);
+
+                    service.Configuration.Add(configItem);
+                }
+            }
+            
+        }
+
+        private static void ParsePortSequence(YamlSequenceNode portSequence, ConfigService service)
+        {
+            foreach (var port in portSequence)
+            {
+                var portString = YamlParser.GetScalarValue(port);
+                var binding = new ConfigServiceBinding();
+                if (portString.Contains(':'))
+                {
+                    var ports = portString.Split(':');
+                    binding.Port = int.Parse(ports[0]);
+                    binding.ContainerPort = int.Parse(ports[1]);
+                }
+                else
+                {
+                    binding.Port = int.Parse(portString);
+                    binding.ContainerPort = int.Parse(portString);
+                }
+
+                // TODO how to specify protocol with docker compose. Using http for now.
+                binding.Protocol = "http";
+                service.Bindings.Add(binding);
+            }
+        }
+
+        private static void ParseVolumes(YamlMappingNode node, ConfigApplication app)
```

Comment:
Nit: This and ParseNetworks are not used yet.

Reply:
Yeah there are a few here that aren't used, but I'd rather get all of the elements defined in this file and we can fill them in later.